src/index.rs

```diff
@@ -396,7 +389,40 @@ impl Index {
   }
 
   pub(crate) fn update(&self) -> Result {
-    Updater::update(self)
+    let mut updater = Updater::new(self)?;
+
+    loop {
```

Comment:
Should this loop be here? 🤔

Reply:
yes, because we need to recover my purging everything in the Updater and starting new

Reply:
Ah, gotcha 👍